File: heeyeon/7주차/1518.Water Bottles.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int answer = 0;
        int full = numBottles;
        int empty = 0;
        
        while(true){
            answer += full;
            empty += full;              //새병이 빈병으로 변경
            full = empty / numExchange; //빈병으로 된 후 새병으로 변경
            empty -= full*numExchange;  //빈병을 새병으로 변경한 후 빈병 수 
            if(full + empty < numExchange){
                answer += full;
                break;
            }
        }
        return answer;
    }
}

File: taewan/week 7/1518. Water Bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        def drink(x):
            return x // numExchange + drink(x // numExchange + x % numExchange) if x // numExchange else 0
        return numBottles + drink(numBottles)